ledger/catchpointwriter_test.go

```diff
@@ -571,6 +572,36 @@ func testNewLedgerFromCatchpoint(t *testing.T, filepath string) *Ledger {
 		return err
 	})
 	require.NoError(t, err)
+
+	balanceTrieStats := func(db db.Accessor) merkletrie.Stats {
+		var stats merkletrie.Stats
+		err = db.Atomic(func(ctx context.Context, tx *sql.Tx) (err error) {
+			committer, err := MakeMerkleCommitter(tx, false)
+			if err != nil {
+				return err
+			}
+			trie, err := merkletrie.MakeTrie(committer, TrieMemoryConfig)
+			if err != nil {
+				return err
+			}
+			stats, err = trie.GetStats()
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+
+		require.NoError(t, err)
+		return stats
+	}
+
+	// Skip invariant check for tests using mocks that do _not_ update
+	// balancesTrie by checking for zero value stats.
+	ws := balanceTrieStats(catchpointWriterReadAccess)
+	if ws != (merkletrie.Stats{}) {
```

Comment:
Why is `ws` allowed to be empty?

Reply:
@jannotti See inline comment above `ws` declaration starting with "Skip invariant check".   I also called out in the PR description that the situation feels kludgey, but I exhausted my available time.  I'm open to discussing live.